From 42f3e299d1af4c76660e767511068f680dfb5ddf Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sun, 28 Feb 2021 18:17:28 +0100
Subject: [PATCH 1/1] efi_loader: console size of vidconsole

If stdout is 'vidconsole' we correctly set the console size.
If stdout is 'vidconsole,serial' the video console is ignored.

We should always evaluate the size of vidconsole if it is the primary
console.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_loader/efi_console.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/efi_loader/efi_console.c b/lib/efi_loader/efi_console.c
index edcfce7bec..c4003554c2 100644
--- a/lib/efi_loader/efi_console.c
+++ b/lib/efi_loader/efi_console.c
@@ -311,7 +311,7 @@ static void query_console_size(void)
 	const char *stdout_name = env_get("stdout");
 	int rows = 25, cols = 80;
 
-	if (stdout_name && !strcmp(stdout_name, "vidconsole") &&
+	if (stdout_name && !strncmp(stdout_name, "vidconsole", 10) &&
 	    IS_ENABLED(CONFIG_DM_VIDEO)) {
 		struct stdio_dev *stdout_dev =
 			stdio_get_by_name("vidconsole");
-- 
2.30.1

